qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -0,0 +1,424 @@
+"""
+Cross resonance Hamiltonian tomography.
+"""
+
+from typing import List, Tuple, Optional, Iterable
+
+import numpy as np
+from qiskit import pulse, circuit, QuantumCircuit
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Backend
+from qiskit.utils import apply_prefix
+
+from qiskit_experiments.framework import BaseExperiment, Options
+from .cr_hamiltonian_analysis import CrossResonanceHamiltonianAnalysis
+
+
+class CrossResonanceHamiltonian(BaseExperiment):
+    r"""Cross resonance Hamiltonian tomography experiment.
+
+    # section: overview
+
+        This experiment assumes the two qubit Hamiltonian in the form
+
+        .. math::
+
+            H = \frac{I \otimes A}{2} + \frac{Z \otimes B}{2}
+
+        where :math:`A` and :math:`B` are Pauli operator :math:`\in {X, Y, Z}`.
+        The coefficient of each Pauli term in the Hamiltonian
+        can be estimated with this experiment.
+
+        This experiment is performed by stretching the pulse duration of the cross resonance tone
+        and measuring the target qubit by projecting onto x, y, and z basis.
+        We will observe the control qubit state dependent (controlled-) Rabi oscillation on the
+        target qubit. The fit for this oscillation tomographically reconstructs the
+        Hamiltonian in the form shown above.
+        Thus the control qubit is prepared in both ground and excited state and
+        the experiment is repeated for both control qubit states.
+        See Ref. [1] for more details.
+
+        More specifically, the following circuits are executed in this experiment.
+
+        .. parsed-literal::
+
+            (X measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├────────
+                 └───┘│  cr_tone(duration) │┌───┐┌─┐
+            q_1: ─────┤1                   ├┤ H ├┤M├
+                      └────────────────────┘└───┘└╥┘
+            c: 1/═════════════════════════════════╩═
+                                                  0
+
+            (Y measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├───────────────
+                 └───┘│  cr_tone(duration) │┌─────┐┌───┐┌─┐
+            q_1: ─────┤1                   ├┤ Sdg ├┤ H ├┤M├
+                      └────────────────────┘└─────┘└───┘└╥┘
+            c: 1/════════════════════════════════════════╩═
+                                                         0
+
+            (Z measurement)
+
+                 ┌───┐┌────────────────────┐
+            q_0: ┤ X ├┤0                   ├───
+                 └───┘│  cr_tone(duration) │┌─┐
+            q_1: ─────┤1                   ├┤M├
+                      └────────────────────┘└╥┘
+            c: 1/════════════════════════════╩═
+                                             0
+
+        The ``X`` gate on the control qubit (``q_0``) depends on the required control state.
```

Comment:
I think it would be better to call the preparation pulse on q_0 `P` (or something else). `X` is a little confusing as a placeholder since `X` is one of the two actual pulses that gets used.

Reply:
Sorry this is due to my poor explanation. I actually indicate the `X` gate because we need control qubit both in excited and ground state. The gate sould be `X` or none. Do you have suggestion on how to describe?

Reply:
okey, I updated the document with `P` gate. https://github.com/Qiskit/qiskit-experiments/pull/385/commits/85e92f5422b012e7dd43f7218e95484d5908a605